build-compiler-rt.sh

```diff
@@ -68,19 +68,17 @@ fi
 
 if command -v ninja >/dev/null; then
     CMAKE_GENERATOR="Ninja"
-    NINJA=1
-    BUILDCMD=ninja
 else
     : ${CORES:=$(nproc 2>/dev/null)}
     : ${CORES:=$(sysctl -n hw.ncpu 2>/dev/null)}
     : ${CORES:=4}
+    export CMAKE_BUILD_PARALLEL_LEVEL=$CORES
```

Comment:
If `$CORES` was set by the user, we should set `CMAKE_BUILD_PARALLEL_LEVEL` for the ninja case as well.

So far, the logic has been that if the user didn't set `$CORES`, with ninja we'd invoke it without any `-j` option, going with whatever ninja's default is (which IIRC isn't exactly the same as `$(nproc)` but a couple processes more), but if `$CORES` was set, we'd set that explicitly.

Reply:
I don't see manually setting CORES adds much value, I understand that you need a better default for make.

Reply:
Actually, it is useful in some cases, to limit the number of parallel jobs. If building e.g. Flang, compiling each object (if building with GCC) file requires vastly more memory than usual. On a machine with many many cores, I've had to set `CORES` to `nproc/4` in some cases to make the Flang build not die by running out of memory.

So to keep the original behaviour of the code, I guess you could have something like outside of the ninja/make block:
```
if [ -n "$CORES" ]; then
   export CMAKE_BUILD_PARALLEL_LEVEL=$CORES
fi
```